### DISCUSSION TITLE
New Build - Watchdog Timeout question

### DISCUSSION THREAD

**Discussion by wmchellis** [picocomputer/community#112] [Q&A] [answered]
Hi All,

Dipping my toes in the 6502 world. Recently built both my Pico Computer and also Lee Harts's 6502 Membership badge. While my badge is working, I am having trouble here. I am getting working VGA Picoout output, but a Watchdog timeout error anytime I interact with RAM. I don't have an oscilloscope (yet...) but I do at least have power going to the 6502. I am new to KiCad so I have not yet run continuity checks on all the lines between the RIA and the 6502. Wondering if I knicked a trace on the board somewhere there.

Wondering two things and would appreciate some insight:

1) What is the Watchdog Timeout function actually watching? I went through the source code looking for that message but couldn't find it. Was hoping that would point me in the direction of particular pair of pins on the RIA / 6502 that I could then test a trace on.

2) What else would you guys look for here? My chips are all socketed and so are my Picos. The VGA output works, so I am assuming it something between the RIA and 6502. I looked for solder shorts but have not found any.

Just want to say I think this is a VERY interesting project, and want to say 'Thank You' to RumbledThumps for it!
Looking forward to getting this one fully functional.

-Bill Chellis

**Comment by rumbledethumps** [comment]
Watchdog errors don't point to any specific problem. The RIA can't see the 6502 running, so it watchdog timeouts. It's probably a missed or cold solder joint. Could be bad, wrong, or missing ICs.

**Reply by wmchellis** [reply]
Thanks for the reply. I thought about changing out my new W65C02 with the older Rockwell 6502 version running over on my badge.  But decided not as I don't think it is a straight swap scenario. The badge has a jumper setting depending on which CPU you put in (Rockwell vs non Rockwell). Don't want to complicate this more than it is already lol.

I will recheck my solder connections.  Will start with the RIA header connections and swap out to a spare PICO I have.

Wondering if you can elaborate on how the RIA knows if the 6502 is running. I looked in the code for the time out to see where that message resided, but so far haven't found it. Curious to know what the RIA looks at to know that the 6502 is working.

Adding in a couple pics of my build. 
![IMG_6729](https://github.com/picocomputer/community/assets/4468400/49ec3be4-6f7a-434a-9b44-9c34c37127b9)
![IMG_6730](https://github.com/picocomputer/community/assets/4468400/ffaab5db-9949-493c-bdca-3a95e5961d76)

I did have this connected earlier to a USB hub & keyboard. That was working also.

I will get it figured out eventually.

-Bill Chellis

**Comment by wmchellis** [comment, marked as answer]
OK, I did find a solder bridge between two pins  (17 & 18) on the RIA. Looks like 18 is a ground pin so 17 was shorting. 
Cleaned up the connection last night and tried out the machine successfully this morning. I can now read and write to RAM with no Watchdog timeout.

Not sure yet what pin 17 on the RIA does specifically, but clearly it matters.

-Bill Chellis
